### ISSUE TITLE
[Layout/RescueEnsureAlignment] Error while parsing rescue inside do/end block

### ISSUE THREAD

**Issue by Dorian** [bbatsov/rubocop#5321]
When creating a new Rails app (with Rails 5.2.0.beta2), it generates a /bin/yarn file that looks like:

```ruby
#!/usr/bin/env ruby
# frozen_string_literal: true

APP_ROOT = File.expand_path('..', __dir__)
Dir.chdir(APP_ROOT) do
  exec "yarnpkg #{ARGV.join(' ')}"
rescue Errno::ENOENT
  warn 'Yarn executable was not detected in the system.'
  warn 'Download Yarn at https://yarnpkg.com/en/docs/install'
  exit 1
end
```

--------

## Expected behavior

Doesn't throw an error.

## Actual behavior

`Layout/RescueEnsureAlignment` throws a parsing error.

## Steps to reproduce the problem

Run rubocop on this example file:

```ruby
[1, 2].each do
  puts "eached"
rescue
  puts "rescued"
end
```

## RuboCop version

```
$ rubocop -V
0.52.0 (using Parser 2.4.0.2, running on ruby 2.5.0 x86_64-darwin14)
```

**Comment by pocke** [comment]
Can you try specifying `TargetRubyVersion` with `.rubocop.yml`?

```yaml
AllCops:
  TargetRubyVersion: 2.5
```


```bash
# without .rubocop.yml
$ rubocop 
Inspecting 1 file
E

Offenses:

test.rb:3:1: E: Lint/Syntax: unexpected token kRESCUE
(Using Ruby 2.1 parser; configure using TargetRubyVersion parameter, under AllCops)
rescue
^^^^^^

1 file inspected, 1 offense detected


# with .rubocop.yml
$ rubocop 
Inspecting 1 file
C

Offenses:

test.rb:1:1: C: Style/FrozenStringLiteralComment: Missing magic comment # frozen_string_literal: true.
[1, 2].each do
^
test.rb:2:8: C: Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
  puts "eached"
       ^^^^^^^^
test.rb:3:1: C: Style/RescueStandardError: Avoid rescuing without specifying an error class.
rescue
^^^^^^
test.rb:4:8: C: Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
  puts "rescued"
       ^^^^^^^^^

1 file inspected, 4 offenses detected
```


And `Layout/RescueEnsureAlignment` cop has another bug, and it is already fixed. See https://github.com/bbatsov/rubocop/issues/5315

**Comment by Dorian** [comment]
@pocke Shouldn't Rubocop default to parsing for ruby 2.5 now? Or to whatever is in `.ruby-version` or the Gemfile's `ruby` directive?

**Comment by pocke** [comment]
>  Or to whatever is in .ruby-version or the Gemfile's ruby directive?

RuboCop sees .ruby-version, but isn't aware of Gemfile's ruby directive now. And i think it is good  that we make RuboCop aware of Gemfile.

> Shouldn't Rubocop default to parsing for ruby 2.5 now? 

I think RuboCop should not parse as Ruby 2.5 by default. Currently RuboCop parses ruby code as Ruby 2.1. It is the oldest version supported by RuboCop.
I think RuboCop should parse ruby code as the oldest version by default. Because it is safest. Ruby 2.5 parser can parse Ruby 2.1 code, but Ruby 2.1 parser can't parse Ruby 2.5 code. So, we can notice parser version mismatch with `Lint/Syntax` offense  by default. But if Ruby 2.5 parser is default, we can't notice.

**Comment by bbatsov** [comment]
@pocke is completely right. I think the current behaviour makes a lot of sense and there's no real reason to change it. I guess the only problem is that some users don't quite understand it. 😄
